### ISSUE TITLE
Is it possible to check if tests are compiling with new-repl command while loading the main project?

### ISSUE THREAD

**Issue by reygoch** [haskell/cabal#5814]
I'm not really interested in loading test suite into my repl (only the library) but I'd like to know if I have any errors in my tests when I reload. Is this (or something similar) possible?

I've tried doing something like `cabal new-repl --enable-tests` in the hopes that my tests will get type checked and compiled in the background but it didn't work.

**Comment by 23Skidoo** [comment]
Does `new-repl test-suite-name` work?

**Comment by reygoch** [comment]
@23Skidoo kinda. Library is present in the context (although it's module names aren't shown in the prompt) but the problem is that test suite is loaded as well which is not ideal. Also, I can't really check other test suites this way.

I actually want to use this with `ghcid` utility so that I can work on my `library` and occasionally switch to my test suites to add some tests without having to close `ghci` and `ghcid` sessions and reload everything.

I have a feeling that this worked at some point but now it doesn't.

**Comment by 23Skidoo** [comment]
Yeah, repl is not smart enough to do that currently. Would it help if there was a `cabal typecheck` command (like `buuld`, but with `-fno-code`) that you could run from time to time?
